src/redux/books/booksSlice.js

```diff
@@ -1,26 +1,38 @@
 import { createSlice } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [],
-  id: '',
-  category: '',
-  title: '',
-  author: '',
-  progres: '',
-  currentChapter: '',
-  chapterNumber: '',
+  books: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
 };
 
 const booksSlice = createSlice({
-  name: 'book',
+  name: 'books',
   initialState,
   reducers: {
-    removeBook: (state, action) => ({
-      books: state.filter((book) => book.id !== action.payload),
-    }),
-    addBook: (state, action) => ({
-      books: [...state.books, action.payload],
-    }),
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.item_id !== action.payload);
+    },
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
   },
 });
```

Comment:
- [ ] [OPTIONAL] One of the backbone and peculiarities of redux is not to mutate the store directly but rather using pure functions to update your store,

To make the Redux slice pure, you should avoid directly mutating the state inside the reducer functions. Here's how you can modify your code to make it pure:

```
const bookSlice = createSlice({
  name: 'book',
  initialState,
  reducers: {
    addBook: (state, action) => {
      return {
        ...state,
        books: [...state.books, action.payload],
      };
    },
    removeBook: (state, action) => {
      const itemId = action.payload;
      return {
        ...state,
        books: state.books.filter((book) => book.item_id !== itemId),
      };
    },
  },
});
```

Reply:
I have done the changes you mentioned, thanks for the kind feedback.